input/system/rds/logs.go

```diff
@@ -88,75 +65,43 @@ func DownloadLogFiles(server *state.Server, logger *util.Logger) (state.Persiste
 			lastMarker = &prevMarker
 		}
 
-		var logFile state.LogFile
-		logFile.UUID = uuid.NewV4()
-		logFile.TmpFile, err = ioutil.TempFile("", "")
+		tmpFile, err := ioutil.TempFile("", "")
 		if err != nil {
 			err = fmt.Errorf("Error allocating tempfile for logs: %s", err)
 			goto ErrorCleanup
 		}
-		logFile.OriginalName = *rdsLogFile.LogFileName
 
 		for {
-			var resp *rds.DownloadDBLogFilePortionOutput
-			resp, err = rdsSvc.DownloadDBLogFilePortion(&rds.DownloadDBLogFilePortionInput{
-				DBInstanceIdentifier: &identifier,
-				LogFileName:          rdsLogFile.LogFileName,
-				Marker:               lastMarker, // This is not set for the initial call, so we only get the most recent lines
-			})
-
+			newBytesWritten, newMarker, additionalDataPending, err := downloadRdsLogFilePortion(rdsSvc, tmpFile, logger, &identifier, rdsLogFile.LogFileName, lastMarker)
 			if err != nil {
-				err = fmt.Errorf("Error downloading logs: %s", err)
-				logFile.Cleanup()
+				tmpFile.Close()
+				os.Remove(tmpFile.Name())
 				goto ErrorCleanup
 			}
 
-			if resp.LogFileData == nil {
-				logger.PrintVerbose("Rds/Logs: No log data in response, skipping")
-				break
-			}
-
-			if len(*resp.LogFileData) > 0 {
-				_, err = logFile.TmpFile.WriteString(*resp.LogFileData)
-				if err != nil {
-					err = fmt.Errorf("Error writing to tempfile: %s", err)
-					logFile.Cleanup()
-					goto ErrorCleanup
-				}
-				bytesWritten += len(*resp.LogFileData)
+			bytesWritten += newBytesWritten
+			if newMarker != nil {
+				lastMarker = newMarker
 			}
 
-			lastMarker = resp.Marker
-
-			if !*resp.AdditionalDataPending {
+			if !additionalDataPending {
 				break
 			}
 		}
 
-		var newLogLines []state.LogLine
-		var newSamples []state.PostgresQuerySample
-
-		readStart := bytesWritten - maxLogParsingSize
-		if readStart < 0 {
-			readStart = 0
-		}
-		_, err = logFile.TmpFile.Seek(int64(readStart), io.SeekStart)
+		buf, tmpFile, err := readLogFilePortion(tmpFile, bytesWritten, logger)
 		if err != nil {
-			err = fmt.Errorf("Error seeking tempfile: %s", err)
-			logFile.Cleanup()
+			tmpFile.Close()
+			os.Remove(tmpFile.Name())
 			goto ErrorCleanup
 		}
 
-		buf := make([]byte, bytesWritten-readStart)
-
-		_, err = io.ReadFull(logFile.TmpFile, buf)
-		if err != nil {
-			err = fmt.Errorf("Error reading %d bytes from tempfile: %s", len(buf), err)
-			logFile.Cleanup()
-			goto ErrorCleanup
-		}
+		newLogLines, newSamples, _ := logs.ParseAndAnalyzeBuffer(string(buf), 0, linesNewerThan, server)
```

Comment:
We could drop the second parameter now, right?

Reply:
Yeah, we can - I'll make a separate cleanup PR, just to keep the change here easier to reason about.